File: basic-java-app/app/src/main/java/com/driouechemed/basicjavaapp/ui/TaskAdapter.java
package com.driouechemed.basicjavaapp.ui;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.driouechemed.basicjavaapp.R;
import com.driouechemed.basicjavaapp.database.entities.Task;
import com.driouechemed.basicjavaapp.databinding.AdapterTaskBinding;

import org.apache.commons.lang3.StringUtils;

import java.util.ArrayList;
import java.util.List;

public class TaskAdapter extends RecyclerView.Adapter<TaskAdapter.ViewHolder> {

    private final List<Task> tasks;

    public TaskAdapter(List<Task> tasks) {
        this.tasks = tasks;
    }

    public TaskAdapter() {
        tasks = new ArrayList<>();
    }

    public void addTask(Task task) {
        tasks.add(task);
        notifyItemInserted(tasks.size() - 1);
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(ViewGroup viewGroup, int viewType) {
        View view = LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.adapter_task, viewGroup, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, final int position) {
        viewHolder.bind(tasks.get(position));
    }

    @Override
    public int getItemCount() {
        return tasks.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {

        AdapterTaskBinding binding;

        public ViewHolder(View view) {
            super(view);
            binding = AdapterTaskBinding.bind(view);
        }

        public void bind(Task task) {
            binding.taskName.setText(task.getTaskName());
            if (StringUtils.isEmpty(task.getTaskDetails())) {
                binding.taskDetails.setVisibility(View.GONE);
            } else {
                binding.taskDetails.setText(task.getTaskDetails());
            }
        }

    }
}

File: basic-kotlin-app/app/src/main/java/com/driouechemed/basickotlinapp/ui/TaskAdapter.kt
package com.driouechemed.basickotlinapp.ui

import android.view.LayoutInflater
import android.view.View
import android.view.View.GONE
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.driouechemed.basickotlinapp.R
import com.driouechemed.basickotlinapp.database.entities.Task
import com.driouechemed.basickotlinapp.databinding.AdapterTaskBinding

class TaskAdapter(private val tasks: ArrayList<Task> = ArrayList()) :
    RecyclerView.Adapter<TaskAdapter.ViewHolder>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(viewGroup.context)
            .inflate(R.layout.adapter_task, viewGroup, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
        viewHolder.bind(tasks[position])
    }

    override fun getItemCount() = tasks.size

    fun addTask(task: Task) {
        tasks.add(task)
        notifyItemChanged(tasks.size - 1)
    }

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {

        private val binding: AdapterTaskBinding = AdapterTaskBinding.bind(view)

        fun bind(task: Task) {
            binding.taskName.text = task.name
            if (task.details.isNullOrEmpty()) {
                binding.taskDetails.visibility = GONE
            } else {
                binding.taskDetails.text = task.details
            }
        }
    }
}
